jwst/pipeline/calwebb_spec3.py

```diff
@@ -100,14 +100,12 @@ def process(self, input):
         self.spectral_leak.save_results = self.save_results
         self.pixel_replace.suffix = 'pixel_replace'
         self.pixel_replace.output_use_model = True
-        self.pixel_replace.save_results = self.save_results
         
         # Overriding the Step.save_model method for the following steps.
         # These steps save intermediate files, resulting in meta.filename
         # being modified. This can affect the filenames of subsequent
         # steps.
         self.outlier_detection.save_model = invariant_filename(self.outlier_detection.save_model)
-        self.pixel_replace.save_model = invariant_filename(self.pixel_replace.save_model)
```

Comment:
Is this line still needed?  Say, if the user manually sets save_results = True, will the subsequent output filenames be affected?

Reply:
I did not find any difference in the filenames (when pixel_replace.save_results=True) in having this line and or removed. 
Since it is not doing any harm maybe we should leave it. Removing it may have unintended effects. 
I will add it back in